File: src/main/java/eu/zavadil/merchmaster/data/shopProduct/shopCategory/ShopProduct.java
package eu.zavadil.merchmaster.data.shopProduct.shopCategory;

import eu.zavadil.merchmaster.data.design.Design;
import eu.zavadil.merchmaster.data.shop.Shop;
import eu.zavadil.merchmaster.data.shopCategory.ShopCategory;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "shop_product")
public class ShopProduct extends ShopProductBase {

	@ManyToOne(optional = false)
	private Shop shop;

	@ManyToOne(optional = false)
	private Design design;

	@ManyToOne(optional = true)
	private ShopCategory category;
}

File: src/ui/src/shared/types/ShopProduct.ts
import {EntityWithName} from "zavadil-ts-common";
import {Shop} from "./Shop";
import {ShopCategory} from "./ShopCategory";
import {Design} from "./Design";

export type ShopProductBase = EntityWithName & {
	visible: boolean;
	creatorProfit: number;
	description?: string | null;
}

export type ShopProduct = ShopProductBase & {
	shop: Shop;
	category?: ShopCategory | null;
	design: Design;
}

export type ShopProductStub = ShopProductBase & {
	shopId: number;
	categoryId?: number | null;
	designId: number;
}
